src/charm.py

```diff
@@ -367,18 +372,25 @@ def _make_env_config(self, domain="svc.cluster.local") -> dict:
         """
         config = self.model.config
         relation = self.model.get_relation("ingress-proxy")
-        if relation:
-            site = relation.data[relation.app]["service-hostname"]
-            svc_name = relation.data[relation.app]["service-name"]
-            svc_port = relation.data[relation.app]["service-port"]
-            backend_site_name = relation.data[relation.app]["service-hostname"]
+        if relation is not None and relation.data[relation.app]:
+            field_missing = False
+            nginx_config_list = ["service-hostname", "service-name", "service-port"]
+            for nginx_config in nginx_config_list:
+                if relation.data[relation.app].get(nginx_config) is None:
+                    field_missing = True
+            if field_missing: return None
```

Comment:
What about `if any(relation.data[relation.app].get(nginx_config) is None for nginx_config in nginx_config_list): return None`